### PR TITLE
New folder structure

### PR DESCRIPTION
**Pull request by lightninglu10**
Folder structure for reference manager

Screen Shot 2023-06-22 at 1 56 43 PM

### CODE REVIEW

File: components/ReferenceManager/references/reference_organizer/ReferenceProjectsUpsertModal.tsx [hunk 1]
```diff
@@ -65,8 +65,14 @@ export default function ReferenceProjectsUpsertModal({
       is_public: isPublic,
       organization: currentOrg?.id,
       project_name: nullthrows(projectName, "Project name may not be null"),
```
Comment (calvinhlee23) [inline components/ReferenceManager/references/reference_organizer/ReferenceProjectsUpsertModal.tsx:67]:
```suggestion
     project: upsertPurpose === "update" ? projectID : null, 
     parent: upsertPurpose === "update" ? null : parent
      is_public: isPublic,
      organization: currentOrg?.id,
      project_name: nullthrows(projectName, "Project name may not be null"),
```

File: components/ReferenceManager/references/reference_organizer/ReferenceProjectsUpsertModal.tsx [hunk 2]
```diff
@@ -65,8 +65,14 @@ export default function ReferenceProjectsUpsertModal({
...
-      project: projectID,
     };
+
+    if (upsertPurpose === "update") {
+      formattedPayload.project = projectID;
+    } else {
+      formattedPayload.parent = projectID;
+    }
```
Comment (calvinhlee23) [inline components/ReferenceManager/references/reference_organizer/ReferenceProjectsUpsertModal.tsx:74]: Are we confident that creating a project will always have an active current project?
Not a fan of if statements like this that mutate already existing objects either. 
Can we do something like what i suggested in above?

File: components/ReferenceManager/references/reference_organizer/context/ReferenceActiveProjectContext.tsx [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { ProjectValue } from "./ReferenceProjectsUpsertContext";
 export type ReferenceActiveProjectContextValueType = {
   activeProject: ProjectValue | null;
   setActiveProject: (proj: ProjectValue) => void;
+  currentOrgProjects: ProjectValue[];
```
Comment (calvinhlee23) [inline components/ReferenceManager/references/reference_organizer/context/ReferenceActiveProjectContext.tsx:7]: nice

**Review by calvinhlee23 — CHANGES_REQUESTED** [review]
Code looks good overall. Made a small comment.
But honestly, I'm not sure how i feel about the UI. 
I'm not really liking the UX in creating a folder and not being able to what kind of citations are there within child projects without having to click to parent than to child.

File: components/ReferenceManager/references/ReferencesContainer.tsx [hunk 4]
```diff
@@ -201,6 +213,10 @@ function ReferencesContainer({
   };
   const inputRef = useRef();
 
+  useEffect(() => {
+    ReactTooltip.rebuild();
+  }, [selectedReferenceIDs]);
+
```
Comment (calvinhlee23) [inline components/ReferenceManager/references/ReferencesContainer.tsx:219]: what is this :o

File: components/ReferenceManager/references/ReferencesContainer.tsx [hunk 5]
```diff
@@ -263,6 +279,10 @@ function ReferencesContainer({
     isFetchingProjects,
   });
 
+  const pluralize = ({ text, length }) => {
```
Comment (calvinhlee23) [inline components/ReferenceManager/references/ReferencesContainer.tsx:282]: can we make a separate file out side.

File: components/ReferenceManager/references/api/updateReferenceCitation.ts
Comment (yattias) [inline components/ReferenceManager/references/api/updateReferenceCitation.ts:1]: Small: If possible, small change is to add "Api" Suffice to both filename and method. That way - it is easier to understand it is doing an "api thing"

File: components/ReferenceManager/references/reference_table/ReferencesTable.tsx [hunk 6]
```diff
@@ -71,6 +95,82 @@ export default function ReferencesTable({
...
+    updateReferenceCitation({
+      payload: {
+        citation_id: rowDragged,
+        // TODO: calvinhlee - create utily functions to format these
+        project: parseInt(moveToFolderId, 10),
+      },
+      onSuccess: () => {
+        setRowDraggedOver();
+      },
+      onError: () => {},
```
Comment (yattias) [inline components/ReferenceManager/references/reference_table/ReferencesTable.tsx:120]: Can we add a `captureEvent` call in here? I think these call will be helpful insights.

```
captureEvent({ msg: "Failed to move folder", error, data: {
  rowDragged,
  moveToFolderId,
  data,
}})
```

**Review by yattias — APPROVED** [review]

**Review by calvinhlee23 — APPROVED** [review]

**Pull request merged**